### PR TITLE
Change FILE* to std::ofstream and std::ifstream objects

### PR DESCRIPTION
**Pull request by uartie**
Fixes 01org/libyami#744

### CODE REVIEW

**Comment by uartie** [comment]
@xuguangxin @YuJiankang @wudping please help to review

**Comment by wudping** [comment]
it's good, for me.

File: tests/v4l2encode.cpp [hunk 1]
```diff
@@ -147,23 +148,23 @@ bool feedOneInputFrame(int fd, int index = -1 /* if index is not -1, simple enqu
...
-    static FILE* fp = NULL;
+    static std::ofstream m_ofs;
 
-    if(!fp) {
-        fp = fopen(outputFileName, "w+");
-        if (!fp) {
+    if(!m_ofs.is_open()) {
+        m_ofs.open((outputFileName, std::ofstream::out | std::ofstream::trunc);
+        if (!m_ofs) {
             fprintf(stderr, "fail to open file: %s\n", outputFileName);
```
Comment (wudping) [inline tests/v4l2encode.cpp]: hi, should this line use std::cerr too?

Comment (uartie) [reply tests/v4l2encode.cpp]: It's a little out of scope for these changes.  I agree, we should replace all fprintf usage... but that should be handled in another patch series.

**Pull request merged**